### DISCUSSION TITLE
It is imposible to overwrite already loeaded configs

### DISCUSSION THREAD

**Discussion by elnur-ibr** [laravel/framework#38467] [Q&A]
- Laravel Version: 8.55.0

### Description:
It is imposible to overwrite already loeaded configs

### Steps To Reproduce
__DIR__.'/../config/config1.php:
```
         return [
                  'key' => 'First time value'
         ]
```

On first Service Provider:
```
         $this->mergeConfigFrom(
            __DIR__.'/../config/config1.php', 'configs'
        );
```

__DIR__.'/../config/config2.php:
```
         return [
                  'key' => 'Second time value'
         ]
```

On first Service Provider:
```
         $this->mergeConfigFrom(
            __DIR__.'/../config/config2.php', 'configs'
        );
```
Second time config loading will not take place due to `Illuminate\Support\ServiceProvider::mergeConfigFrom()` wrong sequence in array_merge:

```
            $config->set($key, array_merge(
                require $path, $config->get($key, [])
            ));
```

**Comment by powellblyth** [comment]
`config('app.log_level')->set('warning');` ?
but ir won't remain.

If you want cross-process configs, you would need to store them in a dataase like mySql, postGres
